src/NetscriptFunctions/Singularity.ts

```diff
@@ -766,7 +767,11 @@ export function NetscriptSingularity(): InternalAPI<ISingularity> {
       helpers.checkSingularityAccess(ctx);
       const facName = getEnumHelper("FactionName").nsGetMember(ctx, _facName);
       const fac = Factions[facName];
-      return [...fac.getInfo().inviteReqs].map((condition) => condition.toJSON());
+      const enemyNotJoinedReqs =
```

Comment:
Another solution to the mismatch would be to add the actual requirements (probably inside the FactionInfo constructor) internally, but there are already many places where enemies are presented differently to all other requirements (e.g. they have their own section in the UI, the invitation modal calls them out separately, etc), and it would also mean basically getting rid of the isBanned flag for factions. Also we would not actually be able to remove the enemies property, since it's still needed for handling removing invites from enemies when joining a faction.

It's not always a problem to provide info differently to the API than it is handled internally. But the question is, why is there a mismatch?

Internally enemies are separate because they do things other requirements don't - joining a faction removes invites from enemy factions. Is this something a player might want to account for? I think so.

For instance imagine a player has already received an invite from faction X, but they have not yet accepted the invite due to e.g. trying to manipulate where faction rep from coding contracts gets distributed. Their script plans on joining faction X and any other lingering invites sometime later this run, but is just holding off for now. Since the invite has already been obtained, the player script should not be looking at invite requirements anymore (e.g. falling below a money requirement doesn't withdraw the invite), but the script should still have a way of knowing that in order to keep the invite for faction X, they need to make sure not to join their enemy faction Y. So I think to really provide players a correct API, enemies do need to be provided in a different manner (and for basically the same reason why they are handled separated internally).